language/tools/move-cli/src/main.rs

```diff
@@ -377,7 +401,12 @@ fn publish(
         }
     }
 
-    OnDiskStateView::create(storage_dir.to_path_buf(), &modules)
+    let state = OnDiskStateView::create(storage_dir.to_path_buf(), &modules)?;
+    if state.save_modules()? {
+        args.generate_interface_files()?;
```

Comment:
Similar comment re: imperativeness here--it seems like we now want to call `generate_interface_files` every time we do `save_modules`, might want to make it impossible to forget to do that.

Reply:
Same as above, will do it in a separate PR.